pkg/apis/sources/v1beta1/ping_conversion.go

```diff
@@ -116,6 +116,13 @@ func (sink *PingSource) ConvertFrom(ctx context.Context, obj apis.Convertible) e
 			// errors already handled in isV1Beta1AnnotationConsistentWithV1Beta2Spec
 			v1beta1Spec := annotations[V1B1SpecAnnotationKey]
 			_ = json.Unmarshal([]byte(v1beta1Spec), &sink.Spec)
+		} else {
+			// Do not set sink.Spec.JsonData as Data/ContentType in v1beta2 may be non-json.
```

Comment:
This is why using an annotation to support round-tripping does not work. 

Anyhow, IMO the most important use case is v1beta1 round-tripping, i.e read/write v1beta1 PingSource, via v1beta2 storage. The data is then always JSON.

Reply:
> Anyhow, IMO the most important use case is v1beta1 round-tripping, i.e read/write v1beta1 PingSource, via v1beta2 storage. The data is then always JSON.

This already works, the current issue is that if v1beta2 PingSource is created, it cannot be properly read by clients using v1beta1/v1alpha2 API (specifically duckv1.SourceSpec)

Reply:
I have removed all annotation related conversion logic given v1beta2 is the storage version, v1beta1 is roundtrip-able with one exception: when jsonData is not marshallable (e.g. `jsondata="hello"`), but the cloudevent on the wire stays the same. 

a minor question is whether or not to clean up these annotations introduced by versions having v1beta1 as storage with a post-install job. I prefer not as it's still useful in case user wants to downgrade from 0.21  (storage v1beta2) -> 0.20 (storage v1beta1) in rare cases.